src/main/java/io/fabric8/elasticsearch/plugin/OpenshiftAPIService.java

```diff
@@ -0,0 +1,139 @@
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.ElasticsearchSecurityException;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.rest.RestStatus;
+
+import com.jayway.jsonpath.JsonPath;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.openshift.client.DefaultOpenShiftClient;
+import okhttp3.MediaType;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+public class OpenshiftAPIService {
+    
+    private static final String ACCEPT = "Accept";
+    private static final String CONTENT_TYPE = "Content-Type";
+    private static final String APPLICATION_JSON = "application/json";
+    private static final Logger LOGGER = Loggers.getLogger(OpenshiftAPIService.class);
+    
+    public String userName(final String token) {
+        Response response = null;
+        try (DefaultOpenShiftClient client = buildClient(token)) {
+            Request okRequest = new Request.Builder()
+                    .url(client.getMasterUrl() + "apis/user.openshift.io/v1/users/~")
+                    .header(ACCEPT, APPLICATION_JSON)
+                    .build();
+            response = client.getHttpClient().newCall(okRequest).execute();
+            final String body = response.body().string();
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("Response: code '{}' {}", response.code(), body);
+            }
+            if(response.code() != RestStatus.OK.getStatus()) {
+                throw new ElasticsearchSecurityException("", RestStatus.fromCode(response.code()));
+            }
+            return JsonPath.read(body,"$.metadata.name");
+        } catch (IOException e) {
+            LOGGER.error("Error retrieving project list", e);
+            throw new ElasticsearchException(e);
+        }        
+    }
+    
+    public Set<String> projectNames(final String token){
+        try (DefaultOpenShiftClient client = buildClient(token)) {
+            Request request = new Request.Builder()
+                .url(client.getMasterUrl() + "apis/project.openshift.io/v1/projects")
+                .header(ACCEPT, APPLICATION_JSON)
+                .build();
+            Response response = client.getHttpClient().newCall(request).execute();
+            if(response.code() != RestStatus.OK.getStatus()) {
+                throw new ElasticsearchSecurityException("Unable to retrieve users's project list", RestStatus.fromCode(response.code()));
+            }
+            return new HashSet<String>(JsonPath.read(response.body().byteStream(),"$.items[*].metadata.name"));
```

Comment:
Is there a way to request only the `metadata` or `metadata.name`?  Purely in the interest of reducing the response payload.